### ISSUE TITLE
The 'X' button from tabs tray is not accessible using voiceover

### ISSUE THREAD

**Issue by SimonBasca** [mozilla-mobile/firefox-ios#11735]
## Steps to reproduce
1. Enable voiceOver
2. Open a few tabs
3. Open tabs tray
4. Try to select the "X" close button 
### Expected behavior
- VoiceOver should read "Close" button
### Actual behavior
- The 'X' button is not accessible using voiceover
### Device & build information
* Device: ?
* Build version: 59568c9ea028c11959f8b02501c6503372385047
### Notes
Attachments:

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FXIOS-4832)

**Comment by TwizzyIndy** [comment]
Hi. I'm just new here. There is a statement that declared the whole `TabCell` as `AccessibilityElement` in `configureWith(tab: Tab, isSelected selected: Bool)` function. As you can see on `L176`. If we really want to make `closeButton` accessible too, we shouldn't declare the whole `TabCell` accessible. And we also need to declare `closeButton` and `titleText` as `accessible`.

https://github.com/mozilla-mobile/firefox-ios/blob/aeef46abeab95708f70551d57458b62e1a15bb98/Client/Frontend/Browser/TabCell.swift#L170-L179

I've made some changes, like the result shown in the screen record below. If this looks to you, I really love to create PR for this. Thanks.

```swift
    // MARK: - Configure tab cell with a Tab
    func configureWith(tab: Tab, isSelected selected: Bool) {
        isSelectedTab = selected

        titleText.text = tab.getTabTrayTitle()
        titleText.accessibilityLabel = getA11yTitleLabel(tab: tab)
        titleText.isAccessibilityElement = true
        titleText.accessibilityHint = .TabTraySwipeToCloseAccessibilityHint
        
        closeButton.isAccessibilityElement = true
        closeButton.accessibilityHint = .CloseButtonTitle
        ....
```


https://user-images.githubusercontent.com/6712795/187856741-392baaa1-72e7-4a9c-b23b-78e2f45b36ce.mp4

**Comment by thatswinnie** [comment]
Thank you so much for looking into this @TwizzyIndy 🙏🏼 You saved us a bunch of time with this! 

Your solution is definitely one that would work. However Apple recommended in one of their WWDC videos to group content like table/collection view cells together instead of reading every element of it. This is exactly what was done here. So I think we should keep it like this.

Looking at the code you posted I don't think that this is a bug at all as the tabs can be closed with a 3 finger swipe gesture just like the accessibility hint suggests. @SimonBasca what do you think? Can this be closed?
